### ISSUE TITLE
PipelineFile lacks publish_type argument at init

### ISSUE THREAD

**Issue by ocehugo** [aodn/python-aodncore#210]
This leads to a lot of set after init within the pipelines.

Looks like there was some particular fencing around this option, and I wonder why this is forced to be of a particular UNSET type and not following from init arguments. For example, why not initialize it to None!?

Practical example:
```
#do this
 x = PipelineFile(...,publish_type=HARVEST_UPLOAD)
#instead of
 x = PipelineFile(...)
 x.publish_type = HARVEST_UPLOAD
```

@lwgordonimos

**Comment by lwgordonimos** [comment]
Agree that recurring pattern is annoying, so I ninja-fixed it in: https://github.com/aodn/python-aodncore/pull/209, which has a couple of usability enhancements to reduce clunky boilerplate code on aodndata, and addresses some of the issues discussed in https://github.com/aodn/python-aodncore/pull/208.

To answer the question re: init arguments, it provides a firm guarantee that property can only contain a PipelineFilePublishType enum value, because it could only be set via the property setter, so we never had to deal with this kind of thing:

```python
x = PipelineFile(..., publish_type=['random', 'type', 'I', 'dont', 'want', 'to', 'deal', 'with'])
x = PipelineFile(..., publish_type=1)
x = PipelineFile(..., publish_type="what is this")
```

Enforcing it at the setter method means code using it doesn't have to deal with weird values or `None` later on.

It's ultimately to support a "pre flight" check to make sure the handler hasn't left any input files hanging around without being assigned a publish type. It means that if a handler wants to ignore a file, it has to *explicitly* ignore it. If a handler receives a file which is unexpected (for example a ZIP structure which has extra files for some reason) we wanted to know about it, and fail cleanly before proceeding with publishing.

https://github.com/aodn/python-aodncore/blob/a8704afe3a30f72dd0ef4ec4e7b909017bf8233d/aodncore/pipeline/handlerbase.py#L762

**Comment by mhidas** [comment]
This was fixed in #209 (deployed as aodncore 1.2.1)
